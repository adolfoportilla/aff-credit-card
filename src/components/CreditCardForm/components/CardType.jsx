import React from "react";

import { ReactComponent as Visa } from "payment-icons/min/flat/visa.svg";
import { ReactComponent as Amex } from "payment-icons/min/flat/amex.svg";
import { ReactComponent as Default } from "payment-icons/min/mono/default.svg";

import "./CardType.scss";

const CardType = ({ type, wrapperClassName }) => {
  switch (type) {
    case "AMEX":
      return (
        <div className={wrapperClassName}>
          <Amex className="card-type" />
        </div>
      );
    case "VISA":
      return (
        <div className={wrapperClassName}>
          <Visa className="card-type" />
        </div>
      );
    default:
      return (
        <div className={wrapperClassName}>
          <Default className="card-type" />
        </div>
      );
  }
};

export default CardType;
