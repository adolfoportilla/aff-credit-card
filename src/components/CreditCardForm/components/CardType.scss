@import '../../../scss/base.scss';

.card-type {
  margin-top: 7.5px;
  margin-left: 7.5px;
  height: 20px;
  width: auto;
}
