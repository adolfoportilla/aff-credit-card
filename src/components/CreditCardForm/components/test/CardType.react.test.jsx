import React from "react";
import { shallow } from "enzyme";

import { ReactComponent as Visa } from "payment-icons/min/flat/visa.svg";
import { ReactComponent as Amex } from "payment-icons/min/flat/amex.svg";
import { ReactComponent as Default } from "payment-icons/min/mono/default.svg";
import CardType from "../CardType";

describe("Input", () => {
  test("renders amex", () => {
    const wrapper = shallow(<CardType type="AMEX" />);

    expect(wrapper.find(Amex)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders visa", () => {
    const wrapper = shallow(<CardType type="VISA" />);

    expect(wrapper.find(Visa)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders default", () => {
    const wrapper = shallow(<CardType type="DEFAULT" />);

    expect(wrapper.find(Default)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
