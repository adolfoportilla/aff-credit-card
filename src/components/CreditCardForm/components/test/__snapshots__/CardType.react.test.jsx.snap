// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders amex 1`] = `ShallowWrapper {}`;

exports[`Input renders default 1`] = `ShallowWrapper {}`;

exports[`Input renders visa 1`] = `ShallowWrapper {}`;
